src/libclipper/src/rpc_service.cpp

```diff
@@ -137,20 +141,36 @@ void RPCService::manage_service(const string address) {
       poll_timeout = 1;
     }
     zmq_poll(items, 1, poll_timeout);
+
     if (items[0].revents & ZMQ_POLLIN) {
       // TODO: Balance message sending and receiving fairly
       // Note: We only receive one message per event loop iteration
       log_info(LOGGING_TAG_RPC, "Found message to receive");
-
       receive_message(socket, connections, connections_containers_map,
                       zmq_connection_id, redis_connection);
     }
     // Note: We send all queued messages per event loop iteration
     send_messages(socket, connections);
+    auto current_time = std::chrono::system_clock::now();
+    if(std::chrono::duration_cast<std::chrono::milliseconds>(current_time - last_check_time_).count() > CONTAINER_EXISTENCE_CHECK_FREQUENCY){
+      check_container_activity();
+      last_check_time_ = current_time;
+    }
   }
   shutdown_service(socket);
 }
 
+void RPCService::check_container_activity(){
+    std::map<const vector<uint8_t>, std::chrono::system_clock::time_point>::iterator it;
+    std::chrono::system_clock::time_point current_time;
+    for(it = receiving_history_.begin(); it!=receiving_history_.end(); it++){
+        current_time = std::chrono::system_clock::now();
+        if(std::chrono::duration_cast<std::chrono::milliseconds>(current_time - it->second).count() > CONTAINER_ACTIVITY_TIMEOUT){
+            log_info(LOGGING_TAG_RPC, "lost contact with a container");
```

Comment:
If we lose contact with the container, we should remove it from `receiving_history_`. Otherwise, this log message continues to fire every 10 seconds.